internal/pkg/term/selector/selector.go

```diff
@@ -196,6 +203,131 @@ type WorkspaceSelector struct {
+// directory or one level down. If no dockerfiles are found, it asks for a custom path.
+func (s *localFileSelector) Dockerfile(selPrompt, notFoundPrompt, selHelp, notFoundHelp string, pathValidator prompt.ValidatorFunc) (string, error) {
+	dockerfiles, err := s.listDockerfiles()
+	if err != nil {
+		return "", err
+	}
+	var sel string
+	dockerfiles = append(dockerfiles, []string{dockerfilePromptUseCustom, DockerfilePromptUseImage}...)
+	sel, err = s.prompt.SelectOne(
+		selPrompt,
+		selHelp,
+		dockerfiles,
+		prompt.WithFinalMessage(dockerfileFinalMsg),
+	)
+	if err != nil {
+		return "", fmt.Errorf("select Dockerfile: %w", err)
+	}
+	if sel != dockerfilePromptUseCustom {
+		return sel, nil
+	}
+	sel, err = s.prompt.Get(
+		notFoundPrompt,
+		notFoundHelp,
+		pathValidator,
+		prompt.WithFinalMessage(dockerfileFinalMsg))
+	if err != nil {
+		return "", fmt.Errorf("get custom Dockerfile path: %w", err)
+	}
+	return sel, nil
+}
+
+// ListDockerfiles returns the list of Dockerfiles within the current
+// working directory and a sub-directory level below. If an error occurs while
+// reading directories, or no Dockerfiles found returns the error.
+func (s *localFileSelector) listDockerfiles() ([]string, error) {
+	wdFiles, err := s.fs.ReadDir(s.workingDirAbs)
+	if err != nil {
+		return nil, fmt.Errorf("read directory: %w", err)
+	}
+	var dockerfiles = make([]string, 0)
+	for _, wdFile := range wdFiles {
+		// Add current file if it is a Dockerfile and not a directory; otherwise continue.
+		if !wdFile.IsDir() {
+			fname := wdFile.Name()
+			if strings.Contains(strings.ToLower(fname), dockerfileName) && !strings.HasSuffix(strings.ToLower(fname), dockerignoreName) {
+				path := filepath.Dir(fname) + "/" + fname
+				dockerfiles = append(dockerfiles, path)
+			}
+			continue
+		}
+
+		// Add sub-directories containing a Dockerfile one level below current directory.
+		subFiles, err := s.fs.ReadDir(wdFile.Name())
+		if err != nil {
+			// swallow errors for unreadable directories
+			continue
+		}
+		for _, f := range subFiles {
+			// NOTE: ignore directories in sub-directories.
+			if f.IsDir() {
+				continue
+			}
+			fname := f.Name()
+			if strings.Contains(strings.ToLower(fname), dockerfileName) && !strings.HasSuffix(strings.ToLower(fname), dockerignoreName) {
+				path := wdFile.Name() + "/" + f.Name()
+				dockerfiles = append(dockerfiles, path)
+			}
+		}
+	}
+	sort.Strings(dockerfiles)
+	return dockerfiles, nil
+}
+
+type configurationSelector struct {
+	prompt prompter
+}
+
+// NewConfigurationSelector constructs a localFconfigurationSelectorileSelector.
+func NewConfigurationSelector(prompt prompter) *configurationSelector {
+	return &configurationSelector{
+		prompt: prompt,
+	}
+}
+
+// Schedule asks the user to select either a rate, preset cron, or custom cron.
+func (s *configurationSelector) Schedule(scheduleTypePrompt, scheduleTypeHelp string, scheduleValidator, rateValidator prompt.ValidatorFunc) (string, error) {
```

Comment:
This is just a change of receiver from `WorkspaceSelector` to `configurationSelector`.

Reply:
I'm not clear on why this is called a `configurationSelector` 🤔 why not a `scheduleSelector`?

Reply:
My thought is that this selector is actually used to ask any configurational questions that do not require any client to read from fs/ssm, etc. `Schedule` happens to be the only existing selector question that is like this.  I wanted to hint it in the name, so that in the future if we want to ask, for example, a `Rate` question, we don't create a `RateSelector` to be the receiver - the receiver could just be this struct. I am happy to change it to `scheduleSelector` though - we can rename it in the future if we load the struct with other questions.

Reply:
Gotcha that makes sense to me! I think my confusion was that we have a package called `config` which is about SSM 😅 so I wasn't sure if `configurationSelector` was meant to be about SSM or something else.

I'd prefer a `scheduleSelector` but I'll leave it up to you!

Reply:
I renamed the selector so that it does not remind folks of SSM!